frontend/src/features/public-form/components/FormPaymentPage/components/PaymentCard.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box, Flex, Text, useDisclosure } from '@chakra-ui/react'
+
+import { ProductItem } from '~shared/types'
+import { centsToDollars, formatCurrency } from '~shared/utils/payments'
+
+import { BxsChevronDown } from '~assets/icons/BxsChevronDown'
+import Button from '~components/Button'
+import Checkbox from '~components/Checkbox'
+import Radio from '~components/Radio'
+
+import PaymentQuantityModal from './PaymentQuantityModal'
+
+const ItemQuantityButton = ({
+  product,
+  onClick,
+}: {
+  product: ProductItem
+  onClick: () => void
+}) => {
+  if (!product.data.multi_qty) {
+    return <></>
+  }
+  return (
+    <Box minWidth="6.75rem" height="2.75rem" onClick={onClick}>
+      <Button
+        rightIcon={<BxsChevronDown fontSize="1.5rem" />}
+        colorScheme="secondary"
+        aria-label="Change"
+        variant="clear"
+      >
+        Qty: {product.quantity}
+      </Button>
+    </Box>
+  )
+}
+
+export const PaymentCard = ({
+  product,
+  onItemChange,
+  isMultiSelect,
+}: {
+  product: ProductItem
+  onItemChange: (
+    productId: string,
+    isSelected: boolean,
+    selectedQuantity: number,
+  ) => void
+  isMultiSelect: boolean
+}) => {
+  const ChoiceElement = isMultiSelect ? Checkbox : Radio
+  const paymentQuantityModalDisclosure = useDisclosure()
+  return (
+    <Box w="100%">
+      <ChoiceElement
+        isChecked={product.selected}
+        onChange={() =>
+          onItemChange(product.data._id, !product.selected, product.quantity)
+        }
+        variant="fullWidth"
+        p="1rem"
+        minH="6.75rem"
+      >
+        <Box flexGrow={1}>
+          <Text textStyle="subhead-1">{product.data.name}</Text>
+          <Text textStyle="h3" mb="0.5rem">
+            {product.data.description}
+          </Text>
+          <Flex alignItems={'center'}>
+            <Box flexGrow={1} as="h2" textStyle="h3">
+              S
+              {formatCurrency(
+                Number(centsToDollars(product.data.amount_cents ?? 0)),
+              )}
```

Comment:
```suggestion
                Number(centsToDollars(product.data.amount_cents)),
              )}
```

Referring to the type `ProductItem`, `amount_cents` is not optional

Reply:
Fixed in 674aee2fc